api/src/resources/entity_def.rs

```diff
@@ -0,0 +1,17 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Hash)]
+pub struct Id(pub String);
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Hash)]
+pub struct Name(pub String);
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+pub struct EntityDef {
+    pub color: u32,
+    pub id: Id,
+    pub inherits_from: Vec<Id>,
+    pub name: Name,
+    pub title: String,
+    pub trainable: bool,
```

Comment:
I see that you added some fields that are returned by the api, but might not always be expected by the api when creating a new `EntityDef` (or updating an existing one).
In particular, when deserialising the json blob passed as argument, some of these fields might be missing, and we don't want to fail on that.

So I would either remove the `id` and `color` fields for now, with a `TODO` to add them back later once we do the nicer refactoring that you have done for `LabelDef`-s, or make these fields optional (and skip serialisation when sending them to the api).

Reply:
Right, good point. I'll update with an equivalent `NewEntityDef` factoring